File: mongo-target/src/main/java/com/japca/mongo/MongoDao.java
package com.japca.mongo;

import org.springframework.data.repository.CrudRepository;

import java.util.List;

/**
 * Created by Jakub Krhovják on 5/10/18.
 */
public interface MongoDao extends CrudRepository<Record, String> {

	List<Record> findByIdentifier(long identifier);
}

File: mongo-target/src/main/kotlin/com/japca/mongo/MongoDao.kt
package com.japca.mongo

import org.springframework.data.repository.reactive.ReactiveCrudRepository
import reactor.core.publisher.Flux

/**
 * Created by Jakub Krhovják on 5/10/18.
 */
interface MongoDao : ReactiveCrudRepository<Record, String> {

    fun findByIdentifier(identifier: Long): Flux<Record>

}
